nips/04.md

```diff
@@ -0,0 +1,53 @@
+NIP-04
+======
+
+Encrypted Direct Text Note
+--------------------------------------
+
+`draft` `optional`
+
+`kind` `4` `encrypted_direct_text_note`, is similar to `kind` `1` `text_note`, but the content is encrypted and a `p` `tag` targets a specific `public-key`.
+
+`content` is base64 encoded, aes-256-cbc encrypted `string` of anything a user wants to write, encrypted using a shared cipher generated by combining the recipients `public-key` with the senders `private-key`.
+
+`tag` `p` is 32-byte hex of the recipents `public-key`. 
+
+In the first note `tag` `e` is a 32-byte sha256 hash of the shared cipher, so the recipient can validate the cipher they generate.
+
+Aditional notes include `tag` `e` 32-byte hex of the message being replied to.
+
+## Initial note
+
+    ```
+    {
+      id: <32-bytes sha256 of the the serialized event data>
+      pubkey: <32-bytes hex-encoded public key of the event creator>,
+      created_at: <unix timestamp>,
+      kind: 4,
+      tags: [
+        ["e", <32-byte sha256 hash of the shared cipher>, <recommended relay URL>],
+        ["p", <32-bytes hex of the recipients publi-key>, <recommended relay URL>],
+        ... // other kinds of tags may be included later
+      ]
+      content: <base-64 encoded aes-256-cbc encrypted string>,
+      sig: <64-bytes signature of the sha256 hash of the serialized event data, which is the same as the "id" field>,
+    }
+    ```
+
+## Additional notes
+
+    ```
+    {
+      id: <32-bytes sha256 of the the serialized event data>
+      pubkey: <32-bytes hex-encoded public key of the event creator>,
+      created_at: <unix timestamp>,
+      kind: 4,
+      tags: [
+        ["e", <32-bytes hex of the id of previous note in the chat>, <recommended relay URL>],
+        ["p", <32-bytes hex of the recipients public-key>, <recommended relay URL>],
```

Comment:
recipient's

Reply:
Many thanks for the proof, much needed!

Reply:
I'll proofread the whole repository and make a PR.